code/modules/chemistry/Reagents-Base.dm

```diff
@@ -842,7 +842,26 @@ datum
 						if (M.traitHolder?.hasTrait("atheist"))
 							boutput(M, "<span class='notice'>You feel insulted... and wet.</span>")
 						else
-							boutput(M, "<span class='notice'>You feel somewhat purified... but mostly just wet.</span>")
+							if (ishuman(M))
+								var/mob/living/carbon/human/H = M
+								if(H.bioHolder?.HasEffect("blood_curse") || H.bioHolder?.HasEffect("blind_curse") || H.bioHolder?.HasEffect("weak_curse") || H.bioHolder?.HasEffect("rot_curse"))
+									H.bioHolder.RemoveEffect("blood_curse")
+									H.bioHolder.RemoveEffect("blind_curse")
+									H.bioHolder.RemoveEffect("weak_curse")
+									H.bioHolder.RemoveEffect("rot_curse")
+									H.visible_message("[H] screams as some black smoke exits their body.")
+									H.emote("scream")
+									random_burn_damage(H, 5)
+									var/turf/T = get_turf(H)
+									if (T && isturf(T))
+										var/datum/effects/system/bad_smoke_spread/S = new /datum/effects/system/bad_smoke_spread/(T)
+										if (S)
+											S.set_up(5, 0, T, null, "#000000")
```

Comment:
That look really flat with #000? Curious if that would look better with a dark grey or is this fine?

Reply:
Too flat, i'll change it to something grey-ish